### PR TITLE
Implement Partial OGIP 32 - switch repository favorites to new implementation

### LINKED ISSUE #4105: OGIP 32: switch repository favorites to new implementation
Since we implement a new generic favorites system, we want the existing repository favorites to also work together with the new system and use the same storage.

- [x] let the portlet use the new `@list-repository-favorites` REST API endpint for getting the user's favorites (#4102)
- [x] make the existing portlet JS use the new `@toggle-favorite`REST API endpoint for toggling favorites (#4102)
- [x] migrate [existing favorites](https://github.com/4teamwork/opengever.core/blob/026471b04ac1b4a1b68275283ad823e40ea36204/opengever/portlets/tree/favorites.py#L35) to the new SQL store (#4100)
- [x] reimplement caching of repository favorites
  - use last max(last modified) of the current user's favorites as cache key
  - put the cache key in the DOM and only load the favorites when the cache key changes
    - maybe use HTTP caching headers
  - when the cachekey is in the DOM, the [etag value](https://github.com/4teamwork/opengever.core/blob/026471b04ac1b4a1b68275283ad823e40ea36204/opengever/portlets/tree/favorites.py#L135) must be kept / adapted, otherwise it can be removed
- [x] remove the new longer used favorites  [storage](https://github.com/4teamwork/opengever.core/blob/026471b04ac1b4a1b68275283ad823e40ea36204/opengever/portlets/tree/favorites.py#L35) and [view](https://github.com/4teamwork/opengever.core/blob/026471b04ac1b4a1b68275283ad823e40ea36204/opengever/portlets/tree/favorites.py#L72) code
- [x] make sure that using the favorites-toggle in the title (#4103) also reloads the favorites portlet
- [x] review python and JS code of the repository favorites feature and remove everything which is not needed anymore

_harvest: `Feature: Favoriten (OGIP 32)`_

### PR DESCRIPTION
**Pull request by elioschmutz**
This PR implements #4105 - switch repository favorites to new implementation.

Overall handling with screen-syncing:
===========================
sync_between_tabs

Caching:
======
caching

closes #4105

### CODE REVIEW

**Comment by elioschmutz** [comment]
@phgross updated 👍

See also my secondlast commit where I'm removing the obsolete `opengever.portlets.tree.enable_favorites` registry-entry.

⚠️ I do not migrate the current registry-value to the new `opengever base interfaces IFavoritesSettings is_feature_enabled` because I assume, that this value is already set correctly. Is that correct?

File: opengever/portlets/tree/favorites.py [hunk 1]
```diff
@@ -139,20 +15,18 @@ def __init__(self, published, request):
...
-        view = repository_root.restrictedTraverse('repository-favorites')
-        return view.get_cache_key()
 
-    def get_repository_roots(self):
-        roots = filter(IRepositoryRoot.providedBy, aq_chain(self.published))
-        if roots:
-            return roots
-        # Avoid catalog query for performance reasons
-        return filter(IRepositoryRoot.providedBy, getSite().objectValues())
+        favorites = FavoriteManager().list_all_repository_favorites(
```
[the remark this answers is not in the document]
Comment (jone) [reply opengever/portlets/tree/favorites.py]: You are right I didn't consider that. 🙈 We could only make it work with less records by storing an extra record with the last change. That is overkill so lets keep it as it is.

**Comment by elioschmutz** [comment]
@jone @phgross I reimplemented the caching. I do no longer use `etag` for caching but `max-age` instead.

caching

File: opengever/portlets/tree/treeportlet.py [hunk 2]
```diff
@@ -115,6 +117,10 @@ def navigation_url(self):
     def favorites_enabled(self):
         return is_favorites_feature_enabled()
 
+    def favorites_cache_key(self):
+        return getMultiAdapter((self.context, self.request),
+                               IETagValue, name='repository-favorites')()
```
Comment (jone) [inline opengever/portlets/tree/treeportlet.py]: Dies könnte eigentlich abgekürzt werden mit:
```python
return FavoriteManager().get_repository_favorites_cache_key(api.user.get_current().getId())
```
Weil eigentlich ist ja der ETagValue-Adapter nicht zuständig für das Generieren des Cachekeys, das macht ja der FavoriteManager. Es wäre direkter, also auch ein bisschen schneller. Ansonsten ist die Direktion nicht so tragisch.

**Pull request merged**